src/providers/ldap/sdap_id_op.c

```diff
@@ -293,16 +303,77 @@ static void sdap_id_conn_data_expire_handler(struct tevent_context *ev,
                                                           struct sdap_id_conn_data);
     struct sdap_id_conn_cache *conn_cache = conn_data->conn_cache;
 
+    if (conn_cache->cached_connection == conn_data) {
+        DEBUG(SSSDBG_TRACE_ALL,
+              "Connection is about to expire, releasing it\n");
+        conn_cache->cached_connection = NULL;
+        sdap_id_release_conn_data(conn_data);
+    }
+}
+
+/* Start idle timer for connection if needed */
+static int sdap_id_conn_data_start_idle_timer(struct sdap_id_conn_data *conn_data)
+{
+    time_t now;
+    int idle_timeout;
+    struct timeval tv;
+
+    now = time(NULL);
+    conn_data->sh->idle_time = now;
+
+    talloc_zfree(conn_data->idle_timer);
+
+    idle_timeout = dp_opt_get_int(conn_data->conn_cache->id_conn->id_ctx->opts->basic,
+                                  SDAP_IDLE_TIMEOUT);
+    DEBUG(SSSDBG_CONF_SETTINGS, "idle timeout is %d\n", idle_timeout);
+
+    memset(&tv, 0, sizeof(tv));
+    tv.tv_sec = now + idle_timeout;
```

Comment:
What is the expected behavior in case `ldap_connection_idle_timeout` set to 0?
Shouldn't this disable feature?